packages/eslint-plugin/tests/rules/no-floating-promises.test.ts

```diff
@@ -1855,5 +1898,47 @@ cursed();
       `,
       errors: [{ line: 3, messageId: 'floatingPromiseArrayVoid' }],
     },
+    {
+      code: `
+declare const x: any;
+function* generator(): Generator<number, void, Promise<number>> {
+  yield x;
+}
+      `,
+      errors: [{ messageId: 'floatingVoid' }],
+    },
+    {
+      code: `
+declare const x: Generator<number, Promise<number>, void>;
+function* generator(): Generator<number, void, void> {
+  yield* x;
+}
+      `,
+      errors: [{ messageId: 'floatingVoid' }],
+    },
+    {
+      code: `
+3 as Promise<number>;
```

Comment:
I'm not a community maintainer, but I'm very much against these tests.
Even, after checking off all the options in the type-checking section, typescript is still giving error on this statement.

[Playground link](https://www.typescriptlang.org/play/?strict=false&noImplicitAny=false&strictNullChecks=false&strictFunctionTypes=false&strictPropertyInitialization=false&strictBindCallApply=false&noImplicitThis=false&noImplicitReturns=false&alwaysStrict=false#code/MwAghgziAKBOD2BbAlhApgHgHYFdECM1YA+AbiA)

Either these tests be removed or should be replaced with valid ts code.

Reply:
Relevant: #8298

+1 that if we can get away with no type error, it'd be better to.

```ts
const value = {};

value as Promise<number>;
```

Reply:
Makes sense. Also, very clever with suggesting the castable `{}` 👍

Reply:
The empty object type has been on my mind a lot lately... 😂